services/worker-manager/src/providers/aws.js

```diff
@@ -254,6 +254,34 @@ class AwsProvider extends Provider {
     }));
   }
 
+  async removeWorker({worker}) {
+    let result;
+    try {
+      result = await this.ec2s[worker.providerData.region].terminateInstances({
+        InstanceIds: [worker.workerId]
+      }).promise();
+    } catch (e) {
+      this.monitor.err(`Error terminating AWS instance: ${e}`);
+
+      return await workerPool.reportError({
+        kind: 'termination-error',
+        title: 'Instance Termination Error',
+        description: e.message,
+        notify: this.notify,
+        WorkerPoolError: this.WorkerPoolError,
+      });
+
+      throw e;
```

Comment:
This will result in a 500 Internal Server Error to the caller.  Is that the best approach?  Should we maybe just return success?

Reply:
this is a failure path tho..... return success even from here? how the caller would know the call has fail then?

Reply:
agreed. just do nothing here

Reply:
Actually, I just realized that in the similar case in `provision` method we don't rethrow. So why do it here.... I removed that line